src/main/scala/supporters/functions/HeapAccessReplacingExpressionTranslator.scala

```diff
@@ -155,15 +155,7 @@ class HeapAccessReplacingExpressionTranslator(symbolConverter: SymbolConverter,
         }
         val args = eFApp.args map (arg => translate(arg))
         val snap = getOrFail(data.fappToSnap, eFApp, context, sorts.Snap, Option.when(Verifier.config.enableDebugging())(PUnknown()))
-        val fapp = App(fun, snap +: args)
```

Comment:
Do you no longer need the distinction because you change the function version to the limited version later when creating the axiom?